src/mlpack/bindings/R/print_input_param.hpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file bindings/R/print_input_param.hpp
+ * @author Yashwant Singh Parihar
+ *
+ * Print the declaration of an input parameter as part of a line in a R
+ * function definition.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_R_PRINT_INPUT_PARAM_HPP
+#define MLPACK_BINDINGS_R_PRINT_INPUT_PARAM_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace bindings {
+namespace r {
+
+/**
+ * Print the declaration of an input parameter as part of a line in a R
+ * function definition.  This doesn't include any commas or anything.
+ */
+template<typename T>
+void PrintInputParam(util::ParamData& d,
+                     const void* /* input */,
+                     void* /* output */)
+{
+  MLPACK_COUT_STREAM << d.name;
+  if (std::is_same<T, bool>::value)
+    MLPACK_COUT_STREAM << "=FALSE";
+  else if (!d.required)
+    MLPACK_COUT_STREAM << "=NA";
```

Comment:
- yes: if you want just Python 3 in Debian/Ubuntu, say `python3`. It rotate to whatever is deemed official, and importantly `pip3` etc use the same

- any object code that ends up _in the R package_ so that the R package check sees it (_i.e._ part of the main internal shared library from the package that R loads) needs to replace `std::cout` with `MLPACK_COUT_STREAM`.  Other code does not

- the link to the docs is great; examples included and rendered is fabulous (one nit: "quickstart" mentions all the other languages, should there be an R link? Maybe that was you reference to "older version use to render this view...")

Reply:
> any object code that ends up in the R package so that the R package check sees it (i.e. part of the main internal shared library from the package that R loads) needs to replace std::cout with MLPACK_COUT_STREAM. Other code does not

Yep, unless I am mistaken, this code shouldn't end up in the R package.  @Yashwants19 of course, correct me if I'm wrong.

> the link to the docs is great; examples included and rendered is fabulous (one nit: "quickstart" mentions all the other languages, should there be an R link? Maybe that was you reference to "older version use to render this view...")

Precisely, if we were to merge this and the website rebuilt, the R quickstart should be linked just fine. :+1:

Reply:
This file and some other files are travelling with `R_option.hpp` in the package, which creates "warning" regarding `std::cout`.

Reply:
Ah, got it, in this case, we have to use `MLPACK_COUT_STREAM`.  Thanks for clarifying! :+1: